pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -0,0 +1,150 @@
+		newPodNICWithMocks         = func(vmi *v1.VirtualMachineInstance) *podNIC {
+			podnic, err := newPodNICWithWithoutInfraConfigurator(vmi, &vmi.Spec.Networks[0], mockNetwork, cacheFactory, nil)
+			Expect(err).ToNot(HaveOccurred())
+			podnic.podInterfaceName = primaryPodInterfaceName
+			podnic.infraConfigurator = mockPodNetworkConfigurator
+			return podnic
+		}
+	)
+	BeforeEach(func() {
+		cacheFactory = fake.NewFakeInMemoryNetworkCacheFactory()
+
+		ctrl = gomock.NewController(GinkgoT())
+		mockNetwork = netdriver.NewMockNetworkHandler(ctrl)
+		mockPodNetworkConfigurator = infraconfigurators.NewMockPodNetworkInfraConfigurator(ctrl)
+	})
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+	Context("when pod networking setup fails", func() {
+		BeforeEach(func() {
+			address1 := &net.IPNet{IP: net.IPv4(1, 2, 3, 4)}
+			address2 := &net.IPNet{IP: net.IPv4(169, 254, 0, 0)}
+			fakeAddr1 := netlink.Addr{IPNet: address1}
+			fakeAddr2 := netlink.Addr{IPNet: address2}
+			mockNetwork.EXPECT().ReadIPAddressesFromLink(primaryPodInterfaceName).Return(fakeAddr1.IP.String(), fakeAddr2.IP.String(), nil)
+			mockNetwork.EXPECT().IsIpv4Primary().Return(true, nil).Times(1)
+			mockPodNetworkConfigurator.EXPECT().DiscoverPodNetworkInterface(primaryPodInterfaceName).Times(1)
+			mockPodNetworkConfigurator.EXPECT().GenerateDHCPConfig().Return(&cache.DHCPConfig{}).Times(1)
+			mockPodNetworkConfigurator.EXPECT().GenerateDomainIfaceSpec().Times(1)
+			mockPodNetworkConfigurator.EXPECT().PreparePodNetworkInterface().Return(fmt.Errorf("podnic_test: forcing prepare networking failure")).Times(1)
+		})
+		It("should return a CriticalNetworkError at phase1", func() {
+
+			domain := NewDomainWithBridgeInterface()
+			vmi := newVMIBridgeInterface("testnamespace", "testVmName")
+
+			api.NewDefaulter(runtime.GOARCH).SetObjectDefaults_Domain(domain)
+
+			podnic := newPodNICWithMocks(vmi)
+			err := podnic.PlugPhase1()
+			Expect(err).To(HaveOccurred(), "SetupPhase1 should return an error")
+
+			_, ok := err.(*neterrors.CriticalNetworkError)
+			Expect(ok).To(BeTrue(), "SetupPhase1 should return an error of type CriticalNetworkError")
+		})
+	})
+	Context("when DHCP startup fails", func() {
+		BeforeEach(func() {
+			mockNetwork.EXPECT().StartDHCP(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(errors.New("podnic_test: forcing failure at DHCP start"))
+		})
+		It("phase2 should panic", func() {
+			testDhcpPanic := func() {
+				domain := NewDomainWithBridgeInterface()
+				vmi := newVMIBridgeInterface("testnamespace", "testVmName")
+				api.NewDefaulter(runtime.GOARCH).SetObjectDefaults_Domain(domain)
+				podnic := newPodNICWithMocks(vmi)
+				//podnic.launcherPID = &launcherPID
+				podnic.storeCachedDomainIface(domain.Spec.Devices.Interfaces[0])
+				podnic.dhcpConfigurator.ExportConfiguration(cache.DHCPConfig{Name: "eth0"})
+				Expect(podnic.PlugPhase2(domain)).To(Succeed())
+			}
+			Expect(testDhcpPanic).To(Panic())
```

Comment:
As we commented the panic gomega mechanism is working fine the problem is that `EnsureDHCPServerStarted` is always failing since it cannot create the locking file.

Reply:
For casual readers:
  - the test is panicking - but for other reasons: `EnsureDHCPServerStarted` is always failing since it cannot create the locking file. 

We've decided to create an issue, mark the test as broken and move along. It'll be fixed as a follow-up.